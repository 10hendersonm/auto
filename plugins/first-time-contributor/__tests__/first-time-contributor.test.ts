import Auto from '@auto-it/core';
import FirstTimeContributor from '../src';

describe('First Time Contributor Plugin', () => {
  test('should do something', async () => {
  });
});
