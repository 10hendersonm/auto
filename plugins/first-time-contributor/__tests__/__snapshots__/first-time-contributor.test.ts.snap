// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Time Contributor Plugin should exclude a past contributor 1`] = `
Array [
  ":tada: This release contains work from a new contributor! :tada:

Thank you, Jeff, for all your work!",
]
`;

exports[`First Time Contributor Plugin should exclude a past contributor by username 1`] = `
Array [
  ":tada: This release contains work from a new contributor! :tada:

Thank you, Jeff123, for all your work!",
]
`;

exports[`First Time Contributor Plugin should include a username if it exists 1`] = `
Array [
  ":tada: This release contains work from new contributors! :tada:

Thanks for all your work!

:heart: Jeff ([@jeff-the-snake](https://github.com/jeff-the-snake))

:heart: Andrew ([@hipstersmoothie](https://github.com/hipstersmoothie))",
]
`;

exports[`First Time Contributor Plugin should include all authors with no past contributors 1`] = `
Array [
  ":tada: This release contains work from new contributors! :tada:

Thanks for all your work!

:heart: Jeff

:heart: Andrew",
]
`;

exports[`First Time Contributor Plugin should include only username if no name exists 1`] = `
Array [
  ":tada: This release contains work from a new contributor! :tada:

Thank you, [@jeff-the-snake](https://github.com/jeff-the-snake), for all your work!",
]
`;

exports[`First Time Contributor Plugin should not include past contributors - email 1`] = `Array []`;

exports[`First Time Contributor Plugin should not include past contributors 1`] = `Array []`;
